sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -185,20 +185,44 @@ def reset_on_fork(restart_thread: true)
 
           def export_batch(span_array, timeout: @exporter_timeout_seconds)
             batch = span_array.map(&:to_span_data)
-            result_code = @export_mutex.synchronize { @exporter.export(batch, timeout: timeout) }
-            report_result(result_code, span_array)
-            result_code
+            result = @export_mutex.synchronize do
+              OpenTelemetry.logger.debug("BatchSpanProcessor#export_batch: exporter=#{@exporter.class.name}")
+              OpenTelemetry.logger.debug("BatchSpanProcessor#export_batch: Exporting batch of #{batch.size} spans with timeout #{timeout}")
```

Comment:
So there's a lot of instances where you're emitting two lines when it could be a single log line invocation, string allocation.

Directly after these logs are emitted, when the export function is called this log line is emitted with the exact same information. 

```rb
 OpenTelemetry.logger.debug("OTLP::HTTP::TraceExporter#export: Called with #{span_data&.size || 0} spans, timeout=#{timeout.inspect}")
```